core/tests/protractor_utils/forms.js

```diff
@@ -34,6 +34,85 @@ var DictionaryEditor = function(elem) {
   };
 };
 
+var GraphEditor = function(graphInputContainer) {
+  if (!graphInputContainer) {
+    throw Error('Please provide Graph Input Container element');
+  }
+  var vertexElement = function(index) {
+    // Would throw incorrect element error if provided incorrect index number.
+    // Node index starts at 0.
+    return graphInputContainer.element(by.css(
+      '.protractor-test-graph-vertex-' + index));
+  };
+
+  var createVertex = function(xOffset, yOffset) {
+    var addNodeButton = graphInputContainer.element(
+      by.css('.protractor-test-Add-Node-button'));
+    addNodeButton.click();
+    // Offsetting from the graph container.
+    browser.actions()
+      .mouseMove(graphInputContainer, {x: xOffset, y: yOffset})
+      .click()
+      .perform();
+  };
+
+  var createEdge = function(vertexIndex1, vertexIndex2) {
+    var addEdgeButton = graphInputContainer.element(
+      by.css('.protractor-test-Add-Edge-button'));
+    addEdgeButton.click();
+    browser.actions()
+      .dragAndDrop(vertexElement(vertexIndex1), vertexElement(vertexIndex2))
+      .perform();
+  };
+  return {
+    setValue: function(graphDict) {
```

Comment:
Pass in vertices and edges as separate args. We don't have a formal Graph domain object here so try to avoid passing around dict blobs.

Reply:
Consider it done!

Reply:
On another note, would an array [vertices, edges] be preferable than a dict here? The framework currently revolves around a  single argument being passed to setValue() for different form Objects.  That was why I went with a dict to begin with since would be better for readability.

Having 2 separate arg would be fine too but I would have to add more codes to the ExplorationEditorMainTab logic solely to support GraphInput interaction.

Reply:
Oh, I see. OK, yeah, let's leave it as you have it, then. Thanks for the explanation!

Reply:
I would have to maneuver around the way testSuite handles submitting answers too.

Heres how testSuite submit answers:  [Answer1, Answer2, ...]. 

This is good for submitting multiple rounds of answers but not for GraphInput, which if converting to 2 different args, [Answer1Info, Answer1Data] 

Overall, my opinion is that dictionary works well for the situation at hand due to already written code.

Reply:
Yep, I agree. I forgot that this was part of a series of extensions which already have certain conventions. I think the extension convention wins here.